jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -545,6 +558,29 @@ public boolean handle()
         return !suspended;
     }
 
+    private void ensureConnectionClose()
+    {
+        _response.getHttpFields().computeField(HttpHeader.CONNECTION, (h, fields) ->
+        {
+            if (fields == null || fields.isEmpty())
+                return HttpConnection.CONNECTION_CLOSE;
+
+            if (fields.stream().anyMatch(f -> f.contains(HttpHeaderValue.CLOSE.asString())))
+            {
+                if (fields.size() == 1)
+                    return fields.get(0);
+
+                return new HttpField(HttpHeader.CONNECTION, fields.stream()
+                    .flatMap(field -> Stream.of(field.getValues()))
+                    .collect(Collectors.joining(", ")));
```

Comment:
Shouldn't duplicates be discarded?

Reply:
Perhaps... but I'm just really leaving the other values exactly as they are, other than amalgamating on to the same header.  If somebody else put a duplicate in there, then that's there problem and I don't want to waste cycles here fixing it.

Reply:
Okay.